src/demo.ts

```diff
@@ -612,12 +613,7 @@ export class DemoFile extends EventEmitter {
       this.emit("tickend", this.currentTick);
       this.emit("end", { error: e });
 
-      // See GH #11: A sizeable proportion of demo files aren't complete.
-      // If we hit a RangeError, just silently swallow it (as the official
-      // game client does)
-      if (!(e instanceof RangeError)) {
-        throw e;
-      }
+      process.emitWarning(e);
```

Comment:
We don't want to emit a warning if the error is a RangeError - these are expected

Reply:
Not a problem, I have reverted the if statement and the comment regarding the RangeError